components/match2/commons/match_group_input.lua

```diff
@@ -456,35 +457,98 @@ end
 function MatchGroupInput.readPlayersOfTeam(match, opponentIndex, teamName, options)
 	options = options or {}
 
-	local maxNumPlayers = options.maxNumPlayers or DEFAULT_MAX_NUM_PLAYERS
 	local opponent = match['opponent' .. opponentIndex]
-	local playersData = Json.parseIfString(opponent.players) or {}
-
 	local players = {}
+	local playersIndex = 0
 
-	for playerIndex = 1, maxNumPlayers do
-		local player = Json.parseIfString(match['opponent' .. opponentIndex .. '_p' .. playerIndex]) or {}
+	local insertIntoPlayers = function(player)
+		if type(player) ~= 'table' or Logic.isEmpty(player) or Logic.isEmpty(player.name) then
+			return
+		end
 
-		player.name = player.name or playersData['p' .. playerIndex]
-			or Variables.varDefault(teamName .. '_p' .. playerIndex)
+		player.name = options.resolveRedirect and mw.ext.TeamLiquidIntegration.resolve_redirect(player.name) or player.name
+		player.name = options.applyUnderScores and player.name:gsub(' ', '_') or player.name
+		player.flag = Flags.CountryName(player.flag)
+		player.displayName = Logic.emptyOr(player.displayName, player.displayname)
```

Comment:
doesn't the left one need to be lowercased?
(since we pass it as is to the storage)

also since you never pass a capitalized one to this i think we can drop the empty or stuff entirely, unless we expect input in `opponnetX_pY` to sometimes use `displayName` as key